keel-orca/src/main/kotlin/com/netflix/spinnaker/keel/orca/OrcaService.kt

```diff
@@ -16,16 +16,18 @@
 package com.netflix.spinnaker.keel.orca
 
 import com.netflix.spinnaker.keel.model.OrchestrationRequest
-import retrofit.http.Body
-import retrofit.http.Headers
-import retrofit.http.POST
+import retrofit.http.*
 
 // TODO Origin needs to be set on executions
+// origin is used for dynamic routing by orca to different clouddriver instances
 interface OrcaService {
 
   @POST("/ops")
   @Headers("Content-Type: application/context+json")
   fun orchestrate(@Body request: OrchestrationRequest): TaskRefResponse
+
+  @GET("/tasks/{id}")
+  fun getTask(@Path("id") id: String): Map<String, Any>
```

Comment:
I'd like to avoid dealing with maps as responses. You could just create a `TaskResponse` data class below like I did for `TaskRefResponse`, defining only the fields we care about:

```kotlin
data class TaskResponse(
  val status: String
  // etc...
)
```

Doing so should make interacting with the client a lot cleaner, and keeps the spread of the evil maps-of-maps pattern isolated to existing services as much as possible.

Reply:
AMAZING